controllers/capabilities/product_controller.go

```diff
@@ -76,7 +79,37 @@ func (r *ProductReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	// Ignore deleted Products, this can happen when foregroundDeletion is enabled
 	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
-	if product.DeletionTimestamp != nil {
+	if product.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(product, productFinalizer) {
+		controllerutil.RemoveFinalizer(product, productFinalizer)
+		err = r.UpdateResource(product)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if product.Status.ID != nil {
+			err = r.removeProduct(product)
+			if err != nil {
+				r.EventRecorder().Eventf(product, corev1.EventTypeWarning, "Failed to delete product", "%v", err)
+				return ctrl.Result{}, err
+			}
+		} else {
+			return ctrl.Result{}, fmt.Errorf("product %s .status.ID is missing, cannot remove product", product.Spec.Name)
```

Comment:
same comment as for the backend controller

Reply:
Thanks